### PR TITLE
Feature 17: Refactor clean architecture

### PR DESCRIPTION
**Pull request by Anwera64**
I've migrated the project to a modularized application and created uses cases that handle business logic.

Advances #17 to 2/3 of the way

### CODE REVIEW

File: data/src/main/java/com/anwera97/data/di.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.anwera97.data
+
+import com.anwera97.data.repository.CompanionRepository
+import com.anwera97.data.repository.ExpenditureRepository
+import com.anwera97.data.repository.TripRepository
+import org.koin.android.ext.koin.androidContext
+import org.koin.dsl.module
+
+val dataModule = module {
```
Comment (josemiZ) [inline data/src/main/java/com/anwera97/data/di.kt:9]: consider migrate this to hilt, it is la leche

File: domain/src/main/java/com/anwera97/domain/mappers/CompanionMapper.kt [hunk 2]
```diff
@@ -39,7 +39,7 @@ private fun toHashMap(results: List<ResultModel>): Map<String, ResultModel> {
     return map
 }
 
-private fun PayerWithExpendituresAndDebtors.getTotalPayment(): Double {
+private fun com.anwera97.data.composedclasses.PayerWithExpendituresAndDebtors.getTotalPayment(): Double {
```
Comment (josemiZ) [inline domain/src/main/java/com/anwera97/domain/mappers/CompanionMapper.kt]: you can add this as a suffix? to make it shorter

File: presentation/src/main/java/com/anwera64/pagodividido/newexpenditure/NewExpenditureActivity.kt [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    private fun setupUi() {
+        setSupportActionBar(binding.toolbar)
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+    }
+
+    private fun loadSpinnerData(nameList: List<CompanionModel>) = with(binding.tiPayer) {
+        val arrayAdapter = ArrayAdapter(context, R.layout.list_metrial_drop_down_item, nameList)
+        setAdapter(arrayAdapter)
+        doOnTextChanged { text, _, _, _ ->
+            nameList.find { companion -> companion.name == text.toString() }
```
Comment (josemiZ) [inline presentation/src/main/java/com/anwera64/pagodividido/newexpenditure/NewExpenditureActivity.kt]: we can rename the companion name to make it more readable

**Review by josemiZ — APPROVED** [review]
LGTM

File: domain/src/main/java/com/anwera97/domain/mappers/CompanionMapper.kt (hunk 2 above)
Comment (Anwera64) [reply domain/src/main/java/com/anwera97/domain/mappers/CompanionMapper.kt]: Sure

File: data/src/main/java/com/anwera97/data/di.kt (hunk 1 above)
Comment (Anwera64) [reply data/src/main/java/com/anwera97/data/di.kt:9]: Lets add it as a ticket

File: presentation/src/main/java/com/anwera64/pagodividido/newexpenditure/NewExpenditureActivity.kt (hunk 3 above)
Comment (Anwera64) [reply presentation/src/main/java/com/anwera64/pagodividido/newexpenditure/NewExpenditureActivity.kt]: What do you mean?

**Pull request merged**